common/flatpak-run.c

```diff
@@ -2941,6 +2942,24 @@ flatpak_run_setup_base_argv (FlatpakBwrap   *bwrap,
   else if (g_file_test ("/var/lib/dbus/machine-id", G_FILE_TEST_EXISTS))
     flatpak_bwrap_add_args (bwrap, "--ro-bind", "/var/lib/dbus/machine-id", "/etc/machine-id", NULL);
 
+  /* As per https://systemd.io/CONTAINER_INTERFACE parse os-release and pass through all
+   * key-value pairs with a service_host_ prefix added to the key */
+  if (g_file_get_contents ("/usr/lib/os-release", &os_release, NULL, NULL))
+    {
+      g_auto(GStrv) os_release_lines = g_strsplit (os_release, "\n", 0);
+      if (os_release_lines)
+        for (int i = 0; os_release_lines[i] != NULL && os_release_lines[i][0] != '\0'; ++i)
+          {
+            g_auto(GStrv) os_release_key_value = g_strsplit (os_release_lines[i], "=", 0);
+            if (!os_release_key_value || !os_release_key_value[0] || os_release_key_value[0][0] == '\0')
+              break;
+            g_autofree gchar *prefixed_key = g_strconcat ("service_host_", os_release_key_value[0], NULL);
```

Comment:
The value might be (and often is) quoted with shell-style quoting. Use `g_shell_unquote()` to undo that.

For example, if the host OS is Debian (where one line of `os-release` is `NAME="Debian GNU/Linux"`), you should set `service_host_NAME` to `Debian GNU/Linux`, not `"Debian GNU/Linux"`.

Reply:
I thought a bit about what to do with quotes, and opted for leaving as-is for 1:1 mapping. But no strong opinion, so changed to unquote.

Reply:
The quotes are not conceptually part of the value of the field. They are part of the encoding of the field to appear in a file that is in shell syntax.

If you `. /etc/os-release`, on (for example) Debian, which has `NAME="Debian GNU/Linux"`, the result is the same as if you ran `NAME="Debian GNU/Linux"`; but if you `export` that, what you get in your environment block is `NAME=Debian GNU/Linux\0`, the same as if you had used `{ "bwrap", "--env", "NAME", "Debian GNU/Linux", ... }` or `env "NAME=Debian GNU/Linux" ...` or similar.

Reply:
That makes sense, thanks. Value is passed through g_shell_unquote now.